### DISCUSSION TITLE
Eco charge / Eco discharge

### DISCUSSION THREAD

**Discussion by svecpetr** [mletenay/home-assistant-goodwe-inverter#141] [Q&A] [answered]
Hi, I see in releases, exactly in January that should be possible to set Eco charge or Eco discharge mode ... not only Eco mode.

I think that I have the latest version of HA, but I did not see or did not know how to use Eco charge or Eco discharge mode

 Home Assistant 2022.11.2 
 Supervisor 2022.10.2
 Operating System 9.3
 Frontend 20221108.0 - latest 

I have HA connected to GoodWe 

GW10K-ET
from GoodWe
Firmware: 04029-08-S11 (02041-17-S00) 

I see in HA only way to change inventer operation mode to:
- General mode
- Off grid mode
- Backup mode
- Eco mode

Please, could anybody help me. Thanks a lot.

**Comment by svecpetr** [comment, marked as answer]
I have already find it out...

Eco charging mode and Eco discharge mode is supported only in HACS installation... So native installation in HA is UNSUPPORTED

**Comment by aichingerk** [comment]
Is there a time line for Eco discharge/charge mode within native installation ?

**Comment by janchlebek** [comment]
> Is there a time line for Eco discharge/charge mode within native installation ?

As far as I know there is no plan yet to move this to HA native integration. @mletenay wants to keep these features in the experimental version only and leave the native one as a more read-only solution (actively modifying inverter parameters without proper knowledge might even lead to damage).

**Reply by aichingerk** [reply]
Is there a possibility to install the HACS integration parallel. I mean the sensor naming or must i deinstall the native integration first ?

**Reply by janchlebek** [reply]
> Is there a possibility to install the HACS integration parallel. I mean the sensor naming or must i deinstall the native integration first ?

No, you have to remove the native before installing experimental. Entities remain same so all your data should be retained.
